custom/icds/views/ccz_hosting.py

```diff
@@ -0,0 +1,267 @@
+            success, error_message = self.form.save()
+            if success:
+                return redirect(self.urlname, domain=self.domain)
+            else:
+                messages.error(request, error_message)
+        return self.get(request, *args, **kwargs)
+
+
+class CCZHostingView(DomainViewMixin, TemplateView):
+    urlname = "ccz_hosting"
+    page_title = ugettext_lazy("CCZ Hosting")
+    template_name = 'icds/ccz_hosting.html'
+
+    @cached_property
+    def ccz_hosting_link(self):
+        return CCZHostingLink.objects.get(identifier=self.identifier)
+
+    def get(self, request, *args, **kwargs):
+        self.identifier = kwargs.get('identifier')
+        try:
+            ccz_hosting_link = self.ccz_hosting_link
+        except CCZHostingLink.DoesNotExist:
+            return HttpResponse(status=404)
+
+        uname, passwd = get_username_and_password_from_request(request)
+        if uname and passwd:
+            if uname == ccz_hosting_link.username and passwd == ccz_hosting_link.get_password:
+                return super(CCZHostingView, self).get(request, *args, **kwargs)
+        # User did not provide an authorization header or gave incorrect credentials.
+        response = HttpResponse(status=401)
+        response['WWW-Authenticate'] = 'Basic realm="%s"' % ''
+        return response
+
+    @property
+    def _page_title(self):
+        return self.ccz_hosting_link.page_title or _("%s CommCare Files" % self.identifier.capitalize())
+
+    def _get_supporting_files(self):
+        return {
+            file_name: reverse('ccz_hosting_download_supporting_files', args=[self.domain, blob_id])
+            for file_name, blob_id in settings.CCZ_FILE_HOSTING_SUPPORTING_FILES.get(self.domain, {}).items()
+        }
+
+    def _get_supporting_docs(self):
+        return {
+            file_name: reverse('ccz_hosting_download_supporting_files', args=[self.domain, blob_id])
+            for file_name, blob_id in settings.CCZ_FILE_HOSTING_SUPPORTING_DOCS.get(self.domain, {}).items()
+        }
+
+    def get_context_data(self, **kwargs):
+        app_names = {app.id: app.name for app in get_brief_apps_in_domain(self.domain, include_remote=True)}
+        return {
+            'page_title': self._page_title,
+            'ccz_hostings': [h.to_json(app_names) for h in CCZHosting.objects.filter(link=self.ccz_hosting_link)],
+            'icds_env': settings.SERVER_ENVIRONMENT in settings.ICDS_ENVS,
+            'supporting_files': self._get_supporting_files(),
+            'supporting_docs': self._get_supporting_docs(),
+        }
+
+
+@login_and_domain_required
+def remove_ccz_hosting(request, domain, hosting_id):
+    try:
+        ccz_hosting = CCZHosting.objects.get(pk=hosting_id)
+        ccz_hosting.delete()
+    except CCZHosting.DoesNotExist:
+        pass
+    return HttpResponseRedirect(reverse(ManageCCZHosting.urlname, args=[domain]))
+
+
+@location_safe
+def download_ccz(request, domain, hosting_id, blob_id):
```

Comment:
I assume this view and the following one intentionally don't require auth, is that correct?

Reply:
ya those have been intentionally skipped. 
Though i should try adding the same authentication as the parent view.